metaflow/plugins/kfp/kfp_utils.py

```diff
@@ -0,0 +1,354 @@
+        RunServiceApi,
+    )
+    import kfp_server_api
+except ImportError:  # Silence import errors in type hint
+    KFPClient = None
+    ApiExperiment = None
+    ApiPipeline = None
+    ApiPipelineVersion = None
+    ApiRun = None
+    RunServiceApi = None
+    kfp_server_api = None
+
+
+def _get_kfp_logger():
+    """Setup logger for KFP plugin
+
+    With expected usage from Jupyter notebook and console in mind,
+    INFO level logs need to show up in Jupyter notebook output cell and consoles.
+    Therefore some default setting below:
+        - Default logging level at logging.DEBUG
+        - Default to a StreamHandler pointing to stdout.
+    Users retain the ability to alter logger behavior using logging module.
+    """
+    kfp_logger = logging.getLogger("metaflow.kfp")
+    if not kfp_logger.hasHandlers():
+        stdout_handler = logging.StreamHandler(sys.stdout)
+        stdout_handler.setLevel(logging.DEBUG)
+        kfp_logger.addHandler(stdout_handler)
+    kfp_logger.setLevel(logging.DEBUG)
+    return kfp_logger
+
+
+logger = _get_kfp_logger()
+
+
+def _get_kfp_client():
+    kfp_client_user = get_username()
+    if KFP_USER_DOMAIN:
+        kfp_client_user += f"@{KFP_USER_DOMAIN}"
+    else:  # FIXME(yunw)(AIP-5686): The KFP_USER_DOMAIN value should be available in flow
+        kfp_client_user += "@zillowgroup.com"
+    return KFPClient(userid=kfp_client_user)
+
+
+def _retry(func: Callable, **kwargs):
+    for attempt in range(1, KFP_CLI_DEFAULT_RETRY + 1):
+        try:
+            return func(**kwargs)
+        except Exception as e:
+            logging.exception(f"Function {func.__name__} failed on attempt {attempt}")
+            if attempt == KFP_CLI_DEFAULT_RETRY:
+                raise e
+
+
+def _get_kfp_run(run_id: str, client: Optional[KFPClient] = None) -> ApiRun:
+    if run_id.startswith("kfp-"):
+        run_id = run_id[len("kfp-") :]  # noqa
+    client: KFPClient = client or _get_kfp_client()
+    return _retry(client.get_run, run_id=run_id).run
+
+
+def run_id_to_url(run_id: str):
+    return posixpath.join(
+        KFP_RUN_URL_PREFIX,
+        "_/pipeline/#/runs/details",
+        run_id,
+    )
+
+
+def to_metaflow_run_id(run_id: str):
+    """Return metaflow run id useful for querying metadata, datastore, log etc."""
+    return f"kfp-{run_id}" if not run_id.startswith("kfp-") else run_id
+
+
+def run_kubeflow_pipeline(
+    pipeline_name: str,
+    kubeflow_namespace: str,
+    triggered_run_name: Optional[str] = None,
+    kubeflow_experiment_name: Optional[str] = None,
+    pipeline_version_id: Optional[str] = None,
+    parameters: Optional[dict] = None,
+    wait_timeout: Union[int, float, datetime.timedelta] = 0,
+    **kwarg,  # Other parameters for wait function
+) -> str:
+    """Trigger KFP flow by pipeline name. See run_kubeflow_pipeline_by_id for more details."""
+    client: KFPClient = _get_kfp_client()
+    kubeflow_pipeline_id: str = client.get_pipeline_id(name=pipeline_name)
```

Comment:
Does this return the latest pipeline?  It'd be good to document

Reply:
Latest pipeline is usually the default pipeline. Will be in the document.